### PR TITLE
Nfs controller publish Changes

### PR DESCRIPTION
**Pull request by VamsiSiddu-7**
# Description
This PR adds the support for nfs controller publish volume

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| https://github.com/dell/csm/issues/763 |

# Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [x] I have verified that new and existing unit tests pass locally with my changes
- [ ] I have not allowed coverage numbers to degenerate
- [ ] I have maintained at least 90% code coverage
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Backward compatibility is not broken

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

- [x] I have Created a new image with the changes and installed the csi-powerflex driver
- [x] I have Executed helm test with volume creation and Controller publish  of nfs,ext4,xfs volumes

### CODE REVIEW

File: service/controller.go [hunk 1]
```diff
@@ -873,15 +891,22 @@ func (s *service) ControllerPublishVolume(
 		fsID := getFilesystemIDFromCsiVolumeID(csiVolID)
 		fs, err := s.getFilesystemByID(fsID, systemID)
 		if err != nil {
-			if strings.EqualFold(err.Error(), sioGatewayVolumeNotFound) || strings.Contains(err.Error(), "must be a hexadecimal number") {
+			if strings.EqualFold(err.Error(), sioGatewayFilesystemNotFound) || strings.Contains(err.Error(), "must be a hexadecimal number") {
 				return nil, status.Error(codes.NotFound,
-					"volume not found")
+					"filesystem not found")
```
Comment (khareRajshree) [inline service/controller.go:896]: update the message wrt to volume, to keep the consistency throughout.

File: service/controller.go [hunk 2]
```diff
@@ -873,15 +891,22 @@ func (s *service) ControllerPublishVolume(
 		fsID := getFilesystemIDFromCsiVolumeID(csiVolID)
 		fs, err := s.getFilesystemByID(fsID, systemID)
 		if err != nil {
-			if strings.EqualFold(err.Error(), sioGatewayVolumeNotFound) || strings.Contains(err.Error(), "must be a hexadecimal number") {
+			if strings.EqualFold(err.Error(), sioGatewayFilesystemNotFound) || strings.Contains(err.Error(), "must be a hexadecimal number") {
```
Comment (khareRajshree) [inline service/controller.go:894]: update the error scenario wrt to volume, to keep the consistency throughout.

File: service/controller.go [hunk 3]
```diff
@@ -873,15 +891,22 @@ func (s *service) ControllerPublishVolume(
...
 				return nil, status.Error(codes.NotFound,
-					"volume not found")
+					"filesystem not found")
 			}
 			return nil, status.Errorf(codes.Internal,
-				"failure checking volume status before controller publish: %s",
+				"failure checking filesystem status before controller publish: %s",
```
Comment (khareRajshree) [inline service/controller.go:899]: update the message wrt to volume, to keep the consistency throughout.

File: service/controller.go [hunk 4]
```diff
@@ -95,6 +98,8 @@ const (
 	removeModeOnlyMe                    = "ONLY_ME"
 	sioGatewayNotFound                  = "Not found"
 	sioGatewayVolumeNotFound            = "Could not find the volume"
+	sioGatewayNFSExportNotFound         = "couldn't find NFS export"
```
Comment (khareRajshree) [inline service/controller.go:101]: If this error code is not being used, we can remove its instance from here.

**Review by khareRajshree — CHANGES_REQUESTED** [review]

File: service/controller.go (hunk 1 above)
Comment (VamsiSiddu-7) [reply service/controller.go]: done

File: service/controller.go (hunk 2 above)
Comment (VamsiSiddu-7) [reply service/controller.go]: done

File: service/controller.go (hunk 3 above)
Comment (VamsiSiddu-7) [reply service/controller.go]: done

File: service/controller.go (hunk 4 above)
Comment (VamsiSiddu-7) [reply service/controller.go]: done

**Comment by VamsiSiddu-7** [comment]
Hi @khareRajshree I have addressed your review comments. please review it again Thanks

**Review by khareRajshree — APPROVED** [review]

**Review by KshitijaKakde — APPROVED** [review]

**Pull request merged**